jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -1465,14 +1418,14 @@ ecma_op_function_construct (ecma_object_t *func_obj_p, /**< Function object */
   {
     case ECMA_OBJECT_TYPE_FUNCTION:
     {
-      arguments_list_p = ecma_op_function_set_construct_flag (arguments_list_p);
+      arguments_list_p = ecma_op_function_args_set_flag (arguments_list_p, ECMA_FUNC_ARG_FLAG_CONSTRUCT);
       ret_value = ecma_op_function_call_simple (func_obj_p, this_arg_value, arguments_list_p, arguments_list_len);
       break;
     }
 #if ENABLED (JERRY_ES2015)
     case ECMA_OBJECT_TYPE_BOUND_FUNCTION:
     {
-      JERRY_ASSERT (!ecma_op_function_has_construct_flag (arguments_list_p));
+      JERRY_ASSERT (!ecma_op_function_args_has_flag (arguments_list_p, ECMA_FUNC_ARG_FLAG_CONSTRUCT));
```

Comment:
Is it okay to only check the `CONSTRUCT` flag? Shouldn't the `SUPER` flag be checked here?

Reply:
Sure!